cmd/kube-controller-manager/app/mizarcontrollers.go

```diff
@@ -152,10 +184,43 @@ func startArktosNetworkController(ctx *ControllerContext, grpcHost string, grpcA
 	}
 	networkClient := arktos.NewForConfigOrDie(&crConfigs)
 	svcKubeClient := clientset.NewForConfigOrDie(netKubeconfigs)
-	informerFactory := externalversions.NewSharedInformerFactory(networkClient, 10*time.Minute)
+	//TO DO: all mizar controllers here should share one informerFactory
+	//       other than every mizar controller starts one informerFactory
+	informerFactory := externalversions.NewSharedInformerFactory(networkClient, 0*time.Minute)
+
+	// Used to create CRDs - VPC or Subnet of tenant
+	dynamicClient := dynamic.NewForConfigOrDie(netKubeconfigs)
+
+	// Used to create mapping to find out GVR via GVK
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(netKubeconfigs)
+	if err != nil {
+		klog.Errorf("when app/mizarcontrollers attempt to start Arktos Network Controller - create discovery Client in error: (%v)", err)
+		return nil, false, err
+	}
+
+	// initialized once for vpcDefaultTemplatePath and subnetDefaultTemplatePath
```

Comment:
It is better to move Line 201-216  into arktos network controller init function.

Reply:
yes, will see whether the function will be created. if easy, I will create it this time.

Reply:
Done and updated in q131172019:CarlXie-poc-2022-01-30-tenant-vpc-support-final